### DISCUSSION TITLE
Allow Python Snippets within Formulas.

### DISCUSSION THREAD

**Discussion by franceme** [mito-ds/mito#33] [Feature Requests]
As a user, I would like the ability to use Python snippets within formulas to create columns.
Using Python for certain functions may make things easier in scenarios.

Ex: Create a column of File Extensions based on a column of Files
Python: os.path.basename(dataframe['File'].str) or something similar

**Comment by naterush** [comment]
This is very doable! Thanks for the suggestion here :-). The example you give makes a ton of sense - using a built-in Python function to do some useful action on one of your columns..

Two questions for you:
1. Do you have another specific example of where you were looking to use some Python? os.path.basename makes a lot of sense, any other real cases you encountered would be super helpful!
2. Were you expecting the os.path.basename to work on the entire column? The Python code as written throws an error, since it expects a single path, and you'd probably have to do a `.apply` or something.

**Comment by naterush** [comment]
My initial thoughts on implementation are:
1. We should be thoughtful about how we handle imports (e.g. os here would need to be in scope). Pandas has useful built ins as well.
2. We shouldn't be restrictive on what this code can return - it would be useful to be able to compute intermediate results that aren't just series...
3. How to handle multi-line statements?
